### PR TITLE
Skip non tensor

### PR DESCRIPTION
**Pull request by anijain2305**
Starting to work on #69 

This remove pytest module related errors while running in PyTorch.

~~~
pytest -v -s test/test_torch.py::TestTorch::test_RNG_after_pickle
~~~

The PR is draft because
* Looking for initial reviews to ensure that direction is good. (cc @jansel)
* Do thorough performance testing with TorchBench on inference and training to ensure that important frames are not missed.
* Do thorough testing with PyTorch tests to see if we need to be more aggressive in skipping more frames.

### CODE REVIEW

File: torchdynamo/convert_frame.py [hunk 1]
```diff
@@ -96,6 +97,28 @@ def _fn(*args, **kwargs):
     return _fn
 
 
+def has_tensor_in_frame(frame):
+    # Check if the passed arguments are of type Tensor
+    for _, value in frame.f_locals.items():
```
Comment (jansel) [inline torchdynamo/convert_frame.py:102]:
```suggestion
    for value in frame.f_locals.values():
```

File: torchdynamo/convert_frame.py [hunk 2]
```diff
@@ -96,6 +97,28 @@ def _fn(*args, **kwargs):
...
+        if value is not None and isinstance(value, torch.Tensor):
+            return True
+
+    # Check if there is global import of torch.*
+    allowed_modules = ["torch"]
+    for _, value in frame.f_globals.items():
```
Comment (jansel) [inline torchdynamo/convert_frame.py:108]: `frame.f_code.co_names` contains a list of all the global variable names this function could read.  

So we don't actually need to iterate over all globals, just the ones that the code could reference.

**Review by jansel — CHANGES_REQUESTED** [review]
A few minor comments, but overall approach looks good.

File: torchdynamo/convert_frame.py [hunk 3]
```diff
@@ -96,6 +97,28 @@ def _fn(*args, **kwargs):
...
+        if inspect.ismodule(value) and value.__name__ in allowed_modules:
```
Comment (jansel) [inline torchdynamo/convert_frame.py:109]: For this test, we might want to reuse some code from `allowed_functions.py`

**Comment by anijain2305** [comment]
Duplicate at https://github.com/pytorch/torchdynamo/pull/248

**Pull request closed without merging**